client/components/data/query-users/index.jsx

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import shallowEqual from 'react-pure-render/shallowEqual';
+
+/**
+ * Internal dependencies
+ */
+import { requestUsers } from 'state/users/actions';
+
+class QueryUsers extends Component {
+	componentWillMount() {
+		this.request();
+	}
+
+	componentDidUpdate( prevProps ) {
+		if (
+			shallowEqual( this.props.usersId, prevProps.usersId ) &&
+			this.props.siteId === prevProps.siteId
+		) {
+			return;
+		}
+
+		this.request();
+	}
+
+	request() {
+		this.props.requestUsers( this.props.siteId, this.props.usersId );
+	}
+
+	render() {
+		return null;
+	}
+}
+
+QueryUsers.propTypes = {
+	requestUsers: PropTypes.func,
+	siteId: PropTypes.number,
+	usersId: PropTypes.arrayOf( PropTypes.number ),
```

Comment:
My current thinking when pluralising multi-entities like that is to keep the entities as small as possible: `user` and `id` instead of a `userId`, and I pluralise those individual entities. It allows to convey a bit more about what it's supposed to contain than pluralising the "combined" entity (`userId`):

- `userId`: contains only one user with one ID
- `usersId`: contains multiple users with one ID each
- `userIds`: contains only one user with multiple IDs
- `usersIds`: contains multiple users with multiple IDs each

In the current use case we expect multiple users, with one ID each, hence why I use `usersId`